hw-2/server.js

```diff
@@ -0,0 +1,92 @@
+import getAutoSuggestUsers from './utils';
+import querySchema, { updateSchema, createSchema } from './validation';
+
+const express = require('express');
+const NodeCache = require('node-cache');
+const uuid = require('uuid');
+const validator = require('express-joi-validation').createValidator();
+
+const cache = new NodeCache();
+const app = express();
+app.use(express.json());
+
+const port = 3000;
+const router = express.Router();
+
+app.set('x-powered-by', false);
+router.get('/users/:id', (req, res) => {
+  const { id } = req.params;
+  if (cache.has(id)) {
+    res.send(cache.get(id));
+  } else {
+    res.send(`User with id: ${id} not found`);
+  }
+});
+router.get('/users/', validator.query(querySchema), (req, res, next) => {
+  try {
+    // throw new Error('Test');
+    const { limit, loginSubstring } = req.query;
+    const data = Object.values(cache.mget(cache.keys()));
+    const filtered = getAutoSuggestUsers(data, loginSubstring, limit);
+    res.send(filtered);
+  } catch (error) {
+    next(error);
+  }
+});
+router.put('/users/:id', validator.body(updateSchema), (req, res, next) => {
+  try {
+    // throw new Error('Test');
+    const { id } = req.params;
+    if (cache.has(id)) {
+      const updated = {
+        ...cache.get(id),
+        ...req.body,
+      };
+      cache.set(id, updated);
+      res.send(updated);
+    } else {
+      res.send(`User with id: ${id} not found`);
```

Comment:
doesn't follow REST best practices

Reply:
changed to status 204 without a body. Is that ok?